tests/test_states.py

```diff
@@ -220,6 +247,394 @@ def test_squeezed_coherent(self):
         self.assertAllAlmostEqual(res.flatten(), res_exact.flatten(), delta=self.tol)
 
 
+class PolyQuadExpectationSingleMode(BaseTest):
+    num_subsystems = 3
+    cutoff = 12
+    hbar = 2
+
+    # qudrature rotation
+    qphi = 0.78
+
+    # construct the expected vector of means and covariance matrix
+    mu = R(qphi).T @ np.array([a.real, a.imag]) * np.sqrt(2*hbar)
+    cov = R(qphi).T @ squeezed_cov(r, phi, hbar=hbar) @ R(qphi)
+
+    def sample_one_mode_normal_expectations(self, mu, cov, func):
+        """Returns the expectation value E(f) and the variance var(f)
+        for some normal distribution X~N(mu, cov).
+
+        Args:
+            mu (array): means vector
+            cov (array): covariance matrix
+            func (function): function acting on the random variables X, P, XP,
+                returning a second order polynomial
+
+        Returns:
+            tuple: tuple of expectation value and variance.
+        """
+        X, P = np.mgrid[-7:7:.01, -7:7:.01]
+        grid = np.dstack((X, P))
+        XP = np.prod(grid, axis=2)
+
+        poly = func(X, P, XP)
+        PDF = multivariate_normal.pdf(grid, mu, cov)
+
+        Ex = simps(simps(poly * PDF, P[0]), X.T[0])
+        ExSq = simps(simps(poly**2 * PDF, P[0]), X.T[0])
+
+        var = ExSq - Ex**2
+
+        return Ex, var
+
+    def setUp(self):
+        """Set up."""
+        super().setUp()
+        self.eng, q = sf.Engine(self.num_subsystems, hbar=self.hbar)
+        self.eng.backend = self.backend
+        self.backend.reset(cutoff_dim=self.cutoff)
+
+        if isinstance(self.backend, backends.TFBackend):
+            raise unittest.SkipTest('The poly_quad_expectation method is not yet supported on the TF backend.')
+
+    def test_no_expectation(self):
+        """Test the case E(0), var(0)"""
+        self.logTestName()
+        state = self.circuit.state()
+
+        A = np.zeros([6, 6])
+        d = np.zeros([6])
+        k = 0
+
+        mean, var = state.poly_quad_expectation(A, d, k)
+        self.assertEqual(mean, 0)
+        self.assertEqual(var, 0)
```

Comment:
We probably want to set these to `assertAlmostEqual` if we change the backend implementations to use `np.isclose`